amy/workshops/tests/test_workshop_staff_searching.py

```diff
@@ -229,6 +229,52 @@ def test_match_on_many_language(self):
 
         self.assertIn(self.ron, response.context['persons'])
         self.assertNotIn(self.harry, response.context['persons'])
+    
+    def test_match_on_one_domain(self):
+        """Ensure people with one particular knowledge domain preference
+        are returned by search."""
+        # prepare knowledge domains
+        self._setUpDomains()
+        # Ron has a humanities knowledge domain
+        self.ron.domains.add(self.humanities)
+        self.ron.save()
```

Comment:
No need to save Ron (Harry would never say that).

The domains exist in M2M relationship, so this call to `.add` here actually is an `INSERT` clause in the intermediary table between `Person` and `KnowledgeDomain` tables. `ron` is not being changed.

Reply:
Oh thank you for pointing that out!